src/v/wasm/wasmtime.cc

```diff
@@ -599,7 +600,13 @@ class wasmtime_engine : public engine {
         // Poll the call future to completion, yielding to the scheduler when
         // the future yields.
         auto start = ss::steady_clock_type::now();
+        // Disable profiling backtraces inside the VM - at the time of writing
+        // backtraces lead to segfaults causing deadlock in Seastar's signal
+        // handlers.
+        auto _ = ss::internal::scoped_disable_profile_temporarily();
```

Comment:
what is the comment mean on this object in seastar tree `This is not reentrant`?

Reply:
It means that it doesn't handle the case of nesting these RAII objects - when being destructed it doesn't set the flag to the original state, but unconditionally turns the flag off.